src/main/java/com/notably/model/suggestion/SuggestionItem.java

```diff
@@ -0,0 +1,19 @@
+package com.notably.model.suggestion;
+
+/**
+ * Represents the instance of the suggested item.
+ */
+interface SuggestionItem {
+    /**
+     * Gets the note path and/ or title displayed in the UI, e.g. "open -t root/cs2103".
+     * @return The note's path and/ or title.
+     */
+    String getDisplayText();
+
+    /**
+     * Executes the action taken after the suggestion item is chosen.
+     * The action will be different depending on the type of SuggestionCommand.
+     * @return The action taken after the suggestion item is chosen.
+     */
+    Runnable run();
```

Comment:
Can I just check with you, which component is supposed to run this Runnable? What exactly does it do? I ask this based on the following understanding that I have: 
1.  The user will use the keyboard to navigate to one the SuggestedItem from the list of items that appears in the SuggestionListview.
2. The user will then press something like "tab" to select that suggestion. This has the effect of triggering some business logic within a view-controller in the UI, that will update the value in the text box (of the commandInputView) with the String of the selected SuggestionItem (retrieved via getDisplayText() method).
3. due to property binding, the CommandInputModel is automatically updated, which will trigger the whole process of correctionEngine->Suggestions generation again. 

Is my understanding correct? If so, a SuggestionItem would not need to have the run() method. Kindly advise :)

Reply:
The `run` method is relevant to your second point. So when a `SuggestionItem item` is chosen, `item.run()` will update the command input in the Suggestion Model using the `setCommandText` method (maybe I should rename this method to `setCommandInputText` for better clarity). Then the UI will just subscribe to this Suggestion Model to update the value in the text box. Sorry I missed `getCommandInputText` method here, will add it in!

Reply:
@HemanshuGandhi your understanding is correct!

This `Runnable` instance exists in each of the `SuggestedItem` and serves the purpose of simplifying the action taken when (2) happens. Each of these `Runnable` encapsulates the business logic required to update the appropriate model to achieve the desired user feedback.

For example, if the "tab" press is supposed to achieve autofill behavior, the `Runnable` encapsulated in that selected `SuggestedItem` represents the logic needed to update the `CommandInputModel` to contain the auto-filled text (which in turn will be propagated to the TextField in the UI)

Another example: if the "tab" press is supposed to achieve an "open note" behavior (like in the search command), the `Runnable` for each `SuggestedItem` encapsulates the logic needed to open the correct notes.

This `Runnable` will be run by the View layer. Basically, this should be run whenever the user presses the action key (e.g. "tab") while selecting a suggestion. All the logic required to update the states of the app would be contained inside this `Runnable`!

Reply:
Alright, this works! Thanks for clarifying.